src/main/kotlin/ar/edu/unahur/obj2/semillasAlViento/Parcela.kt

```diff
@@ -2,44 +2,37 @@ package ar.edu.unahur.obj2.semillasAlViento
 
 class Parcela(val ancho: Int, val largo: Int, val horasSolPorDia: Int) {
   val plantas = mutableListOf<Planta>()
-  var cantidadPlantas = 0
 
   fun superficie() = ancho * largo
-  fun cantidadMaximaPlantas() =
-    if (ancho > largo) ancho * largo / 5 else ancho * largo / 3 + largo
+
+  fun cantidadMaximaPlantas() = if (ancho > largo) superficie() / 5 else superficie() / 3 + largo
+
+  fun tieneComplicaciones() = plantas.any { it.horasDeSolQueTolera() < horasSolPorDia }
+
+  fun puedePlantar(planta: Planta) =
+    plantas.size < this.cantidadMaximaPlantas()
+    && horasSolPorDia <= planta.horasDeSolQueTolera() + 2
 
   fun plantar(planta: Planta) {
-    if (cantidadPlantas == this.cantidadMaximaPlantas()) {
-      println("Ya no hay lugar en esta parcela")
-    } else if (horasSolPorDia > planta.horasDeSolQueTolera() + 2) {
-      println("No se puede plantar esto acá, se va a quemar")
-    } else {
-      plantas.add(planta)
-      cantidadPlantas += 1
-    }
+    if (this.puedePlantar(planta)) plantas.add(planta)
+    else throw Exception("Ya no hay lugar en esta parcela o la planta se va a quemar")
   }
+
+  fun esSemillera() = plantas.isNotEmpty() && plantas.all { it.daSemillas() }
+
+  fun espacioLibre() = this.cantidadMaximaPlantas() - this.plantas.size
 }
 
 class Agricultora(val parcelas: MutableList<Parcela>) {
```

Comment:
Las parcelas no tenían que cambiar
```suggestion
class Agricultora(val parcelas: List<Parcela>) {
```